mycore-base/src/main/java/org/mycore/access/facts/MCRFactsAccessSystem.java

```diff
@@ -139,15 +139,18 @@ public boolean checkPermission(final String checkID, String permission, MCRUserI
         } else {
             if (MCRObjectID.isValid(checkID)) {
                 MCRObjectID oid = MCRObjectID.getInstance(checkID);
-                target = "metadata";
-
-                if ("derivate".equals(oid.getTypeId())) {
-                    facts.add(new MCRObjectIDFact("derid", checkID, oid));
-                    target = "files";
-                    MCRDerivate deriv = MCRMetadataManager.retrieveMCRDerivate(oid);
-                    facts.add(new MCRObjectIDFact("objid", checkID, deriv.getOwnerID()));
+                target = "derivate".equals(oid.getTypeId()) ? "files" : "metadata";
+                
+                if (MCRMetadataManager.exists(oid)) {
+                    if ("derivate".equals(oid.getTypeId())) {
+                        facts.add(new MCRObjectIDFact("derid", checkID, oid));
+                        MCRDerivate deriv = MCRMetadataManager.retrieveMCRDerivate(oid);
```

Comment:
be positive and pretend that all things go well and catch Exception if you need to.
Please consider using org.mycore.datamodel.metadata.MCRMetadataManager#getObjectId to get MCRObjectID of derivate owner.

I propose a cache time of 10 minutes for this as it is usually enough.